av1an-core/src/broker.rs

```diff
@@ -269,52 +269,60 @@ impl<'a> Broker<'a> {
             tpl_crash_workaround = true;
           }
         } else {
+          if current_pass == passes {
+            let encoded_frames = ffmpeg::num_frames(chunk.output().as_ref()).unwrap();
+
+            if encoded_frames == chunk.frames {
+              let enc_time = st_time.elapsed();
+              let fps = encoded_frames as f64 / enc_time.as_secs_f64();
+
+              let progress_file = Path::new(&self.project.temp).join("done.json");
+              get_done().done.insert(
+                chunk.name(),
+                DoneChunk {
+                  frames: encoded_frames,
+                  size_bytes: Path::new(&chunk.output())
+                    .metadata()
+                    .expect("Unable to get size of finished chunk")
+                    .len(),
+                },
+              );
+
+              let mut progress_file = File::create(&progress_file).unwrap();
+              progress_file
+                .write_all(serde_json::to_string(get_done()).unwrap().as_bytes())
+                .unwrap();
+
+              update_progress_bar_estimates(
+                frame_rate,
+                self.project.frames,
+                self.project.verbosity,
+                audio_size_bytes.load(atomic::Ordering::SeqCst),
+              );
+
+              debug!(
+                "finished chunk {:05}: {} frames, {:.2} fps, took {:.2?}",
+                chunk.index, chunk.frames, fps, enc_time
+              );
+            } else {
+              error!(
+                "finished chunk: FRAME MISMATCH: chunk {}: {}/{} (actual/expected frames)",
```

Comment:
Perhaps this should be a warning message instead of an error since we are still restarting the chunk? I think this might be confusing to some users, what do you think though?

Reply:
That's fair, it looks like in the other places we warn unless it's at max retries and bails.